[2025-01-17T16:54:33.216+0000] {taskinstance.py:1159} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: etl_process_with_excel_report.load_data scheduled__2025-01-16T00:00:00+00:00 [queued]>
[2025-01-17T16:54:33.287+0000] {taskinstance.py:1159} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: etl_process_with_excel_report.load_data scheduled__2025-01-16T00:00:00+00:00 [queued]>
[2025-01-17T16:54:33.289+0000] {taskinstance.py:1361} INFO - Starting attempt 9 of 9
[2025-01-17T16:54:33.376+0000] {taskinstance.py:1382} INFO - Executing <Task(PythonOperator): load_data> on 2025-01-16 00:00:00+00:00
[2025-01-17T16:54:33.510+0000] {standard_task_runner.py:57} INFO - Started process 1586 to run task
[2025-01-17T16:54:33.527+0000] {standard_task_runner.py:84} INFO - Running: ['***', 'tasks', 'run', 'etl_process_with_excel_report', 'load_data', 'scheduled__2025-01-16T00:00:00+00:00', '--job-id', '50', '--raw', '--subdir', 'DAGS_FOLDER/db_to_xls.py', '--cfg-path', '/tmp/tmp9o8rrlqg']
[2025-01-17T16:54:33.538+0000] {standard_task_runner.py:85} INFO - Job 50: Subtask load_data
[2025-01-17T16:54:34.055+0000] {task_command.py:415} INFO - Running <TaskInstance: etl_process_with_excel_report.load_data scheduled__2025-01-16T00:00:00+00:00 [running]> on host 4e4dfd1b49e6
[2025-01-17T16:54:34.621+0000] {taskinstance.py:1660} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='***' AIRFLOW_CTX_DAG_ID='etl_process_with_excel_report' AIRFLOW_CTX_TASK_ID='load_data' AIRFLOW_CTX_EXECUTION_DATE='2025-01-16T00:00:00+00:00' AIRFLOW_CTX_TRY_NUMBER='9' AIRFLOW_CTX_DAG_RUN_ID='scheduled__2025-01-16T00:00:00+00:00'
[2025-01-17T16:54:34.998+0000] {base.py:73} INFO - Using connection ID 'local_***' for task execution.
[2025-01-17T16:54:35.229+0000] {sql.py:418} INFO - Running statement: 
        CREATE TABLE IF NOT EXISTS sprout.user_profiling (
            user_id INT,
            name TEXT,
            timestamp TIMESTAMP,
            membership_type TEXT,
            membership_purchase_status TEXT,
            mdr_percentage FLOAT,
            first_transaction TIMESTAMP,
            invoice_total_count INT,
            invoice_total_amount FLOAT
        );
        , parameters: None
[2025-01-17T16:54:35.337+0000] {base.py:73} INFO - Using connection ID 'local_***' for task execution.
[2025-01-17T16:54:35.376+0000] {sql.py:418} INFO - Running statement: 
        CREATE TABLE IF NOT EXISTS sprout.transaction_totals_per_type (
            timestamp TIMESTAMP,
            membership_type TEXT,
            invoice_total_count INT,
            invoice_total_amount FLOAT
        );
        , parameters: None
[2025-01-17T16:54:35.454+0000] {base.py:73} INFO - Using connection ID 'local_***' for task execution.
[2025-01-17T16:54:35.484+0000] {sql.py:418} INFO - Running statement: 
        CREATE TABLE IF NOT EXISTS sprout.daily_activity_report (
            timestamp TIMESTAMP,
            invoice_total_count INT,
            invoice_total_amount FLOAT
        );
        , parameters: None
[2025-01-17T16:54:35.559+0000] {base.py:73} INFO - Using connection ID 'local_***' for task execution.
[2025-01-17T16:54:35.591+0000] {sql.py:418} INFO - Running statement: 
        CREATE TABLE IF NOT EXISTS sprout.transactions_with_mdr (
            transaction_id INT,
            user_id int,
            transaction_amount float,
            timestamp TIMESTAMP,
            name TEXT,
            email TEXT,
            phone TEXT,
            membership_id FLOAT,
            membership_type TEXT,
            purchase_date DATE,
            expiry_date DATE,
            mdr_id FLOAT,
            mdr_percentage FLOAT,
            membership_purchase_status TEXT,
            effective_amount FLOAT
        );
        , parameters: None
[2025-01-17T16:54:35.771+0000] {base.py:73} INFO - Using connection ID 'local_***' for task execution.
[2025-01-17T16:54:35.904+0000] {sql.py:515} INFO - Done loading. Loaded a total of 28 rows into sprout.user_profiling
[2025-01-17T16:54:35.990+0000] {base.py:73} INFO - Using connection ID 'local_***' for task execution.
[2025-01-17T16:54:36.097+0000] {sql.py:515} INFO - Done loading. Loaded a total of 10 rows into sprout.transaction_totals_per_type
[2025-01-17T16:54:36.181+0000] {base.py:73} INFO - Using connection ID 'local_***' for task execution.
[2025-01-17T16:54:36.277+0000] {sql.py:515} INFO - Done loading. Loaded a total of 6 rows into sprout.daily_activity_report
[2025-01-17T16:54:36.370+0000] {base.py:73} INFO - Using connection ID 'local_***' for task execution.
[2025-01-17T16:54:36.428+0000] {db_to_xls.py:228} ERROR - Error during data loading: can't adapt type 'NAType'
[2025-01-17T16:54:36.430+0000] {taskinstance.py:1943} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.9/site-packages/airflow/operators/python.py", line 192, in execute
    return_value = self.execute_callable()
  File "/home/airflow/.local/lib/python3.9/site-packages/airflow/operators/python.py", line 209, in execute_callable
    return self.python_callable(*self.op_args, **self.op_kwargs)
  File "/opt/airflow/dags/db_to_xls.py", line 220, in load_data
    postgres_hook.insert_rows(
  File "/home/airflow/.local/lib/python3.9/site-packages/airflow/providers/common/sql/hooks/sql.py", line 509, in insert_rows
    cur.execute(sql, values)
psycopg2.ProgrammingError: can't adapt type 'NAType'
[2025-01-17T16:54:36.529+0000] {taskinstance.py:1400} INFO - Marking task as FAILED. dag_id=etl_process_with_excel_report, task_id=load_data, execution_date=20250116T000000, start_date=20250117T165433, end_date=20250117T165436
[2025-01-17T16:54:36.564+0000] {standard_task_runner.py:104} ERROR - Failed to execute job 50 for task load_data (can't adapt type 'NAType'; 1586)
[2025-01-17T16:54:36.711+0000] {local_task_job_runner.py:228} INFO - Task exited with return code 1
[2025-01-17T16:54:36.768+0000] {taskinstance.py:2784} INFO - 0 downstream tasks scheduled from follow-on schedule check
